src/plugins/finalcutpro/inspector/video.lua

```diff
@@ -144,25 +145,43 @@ end
 local function doStabilizationMethod(value)
     local timeline = fcp:timeline()
     local timelineContents = timeline:contents()
+    local stabilization = fcp:inspector():video():stabilization()
     local method = fcp:inspector():video():stabilization():method()
 
-    return Do(function()
+    return If(function()
         --------------------------------------------------------------------------------
         -- Make sure at least one clip is selected:
         --------------------------------------------------------------------------------
         local clips = timelineContents:selectedClipsUI()
         if clips and #clips == 0 then
-            log.ef("Set Blend Mode Failed: No clips selected.")
-            tools.playErrorSound()
+            displayMessage(i18n("noSelectedClipsInTimeline"))
             return false
+        else
+            return true
         end
-
-        return Do(method:doSelectValue(value))
-        :Then(WaitUntil(method):Is(value):TimeoutAfter(2000))
-        :Then(true)
-    end)
+    end):Is(true):Then(
+        If(stabilization:doShow()):Is(true)
+        :Then(
+            If(function() return stabilization:isShowing() end):Is(true)
```

Comment:
Same here...
```suggestion
            If(function() return stabilization:isShowing() end)
```

Reply:
In fact, I'm pretty sure that `isShowing` in this case is a `cp.prop`, so we can just pass it in directly, like so:
```suggestion
            If(stabilization.isShowing)
```

Reply:
The reason we can do this is that the value of `stabilization.isShowing` will update dynamically. The main trick with Rx is that you generally don't want to resolve the values being passed into the various `If` or `Do` functions immediately - they will get resolved when the final statement is actually executed (which may happen multiple times in its lifetime). `cp.prop` values are always up-to-date, so they can be checked for updates on demand.